src/main/java/com/hrznstudio/galacticraft/block/special/walkway/PipeWalkway.java

```diff
@@ -0,0 +1,276 @@
+                base = VoxelShapes.union(
+                        base,
+                        Block.createCuboidShape(6.0D, 2.0D, 6.0D, 10.0D, 6.0D, 10.0D),
+                        Block.createCuboidShape(0.0D, 0.0D, 0.0D, 16.0D, 3.0D, 16.0D));
+                break;
+            case NORTH:
+                base = VoxelShapes.union(
+                        base,
+                        Block.createCuboidShape(6.0D, 6.0D, 2.0D, 10.0D, 10.0D, 6.0D),
+                        Block.createCuboidShape(0.0D, 0.0D, 0.0D, 16.0D, 16.0D, 3.0D));
+                break;
+            case SOUTH:
+                base = VoxelShapes.union(
+                        base,
+                        Block.createCuboidShape(6.0D, 6.0D, 10.0D, 10.0D, 10.0D, 14.0D),
+                        Block.createCuboidShape(0.0D, 0.0D, 13.0D, 16.0D, 16.0D, 16.0D));
+                break;
+            case EAST:
+                base = VoxelShapes.union(
+                        base,
+                        Block.createCuboidShape(10.0D, 6.0D, 6.0D, 14.0D, 10.0D, 10.0D),
+                        Block.createCuboidShape(13.0D, 0.0D, 0.0D, 16.0D, 16.0D, 16.0D));
+                break;
+            case WEST:
+                base = VoxelShapes.union(
+                        base,
+                        Block.createCuboidShape(2.0D, 6.0D, 6.0D, 6.0D, 10.0D, 10.0D),
+                        Block.createCuboidShape(0.0D, 0.0D, 0.0D, 3.0D, 16.0D, 16.0D));
+                break;
+        }
+
+        return base;
+    }
+
+    @Override
+    public VoxelShape getOutlineShape(BlockState state, BlockView world, BlockPos pos, ShapeContext context) {
+        return getShape(state);
+    }
+
+    @Override
+    public VoxelShape getCollisionShape(BlockState state, BlockView world, BlockPos pos, ShapeContext context) {
+        return getShape(state);
+    }
+
+    private VoxelShape getShape(BlockState state) {
+        int index = getShapeIndex(state);
+        if (shape[index] != null) {
+            return shape[index];
+        }
+        return shape[index] = createShape(state.get(FACING));
+    }
+
+    private int getShapeIndex(BlockState state) {
+        return this.SHAPE_INDEX_CACHE.computeIntIfAbsent(state, (blockState) -> {
+            int i = 0;
+            if (blockState.get(FACING).equals(Direction.NORTH))
+                i |= getFacingMask(Direction.NORTH);
+            if (blockState.get(FACING).equals(Direction.SOUTH))
+                i |= getFacingMask(Direction.SOUTH);
+            if (blockState.get(FACING).equals(Direction.EAST))
+                i |= getFacingMask(Direction.EAST);
+            if (blockState.get(FACING).equals(Direction.WEST))
+                i |= getFacingMask(Direction.WEST);
+            if (blockState.get(FACING).equals(Direction.UP))
+                i |= getFacingMask(Direction.UP);
+            if (blockState.get(FACING).equals(Direction.DOWN))
+                i |= getFacingMask(Direction.DOWN);
+            return i;
```

Comment:
```suggestion
            return getFacingMask(blockState.get(FACING));
```
I don't totally understand what's happening here, but I think the cache can just be Direction -> int, since that's the only thing where ted about the state.

Probably the same thing with the other walkway classes.

Reply:
You're probably right, I was just simplifying the old code where it also used the connections to the block.